controllers/metal3remediation_controller_test.go

```diff
@@ -190,230 +224,373 @@ func setReconcileNormalRemediationExpectations(ctrl *gomock.Controller,
+			RemediationPhase:    infrav1.PhaseWaiting,
+			IsFinalizerSet:      true,
+			IsPowerOffRequested: false,
+			IsPoweredOn:         true,
+			IsNodeBackedUp:      true,
+			IsNodeDeleted:       true,
+			IsTimedOut:          true,
+		}),
+		Entry("Should restore node and clean up and requeue", reconcileNormalRemediationTestCase{
+			ExpectError:         false,
+			ExpectRequeue:       true,
+			RemediationPhase:    infrav1.PhaseWaiting,
+			IsFinalizerSet:      true,
+			IsPowerOffRequested: false,
+			IsPoweredOn:         true,
+			IsNodeBackedUp:      true,
+			IsNodeDeleted:       false,
+			IsTimedOut:          false,
+		}),
+		Entry("Should skip restore node if forbidden and clean up and requeue", reconcileNormalRemediationTestCase{
+			ExpectError:         false,
+			ExpectRequeue:       true,
+			RemediationPhase:    infrav1.PhaseWaiting,
+			IsFinalizerSet:      true,
+			IsPowerOffRequested: false,
+			IsPoweredOn:         true,
+			IsNodeBackedUp:      true,
+			IsNodeDeleted:       true,
+			IsNodeForbidden:     true,
+			IsTimedOut:          false,
+		}),
+		Entry("Should check if retry limit is reached, and restart remediation if false, and then requeue", reconcileNormalRemediationTestCase{
+			ExpectError:         false,
+			ExpectRequeue:       true,
+			RemediationPhase:    infrav1.PhaseWaiting,
+			IsFinalizerSet:      true,
+			IsPowerOffRequested: false,
+			IsPoweredOn:         true,
+			IsNodeBackedUp:      true,
+			IsNodeDeleted:       true,
+			IsTimedOut:          true,
+			IsRetryLimitReached: false,
+		}),
+		Entry("Should check if retry limit is reached, and trigger machine deletion if true, and don't requeue", reconcileNormalRemediationTestCase{
+			ExpectError:         false,
+			ExpectRequeue:       false,
+			RemediationPhase:    infrav1.PhaseWaiting,
+			IsFinalizerSet:      true,
+			IsPowerOffRequested: false,
+			IsPoweredOn:         true,
+			IsNodeBackedUp:      true,
+			IsNodeDeleted:       true,
+			IsTimedOut:          true,
+			IsRetryLimitReached: true,
+		}),
+		Entry("Should not requeue for Phase Deleting", reconcileNormalRemediationTestCase{
+			ExpectError:      false,
+			ExpectRequeue:    false,
+			RemediationPhase: infrav1.PhaseDeleting,
+		}),
+		Entry("Should not requeue for Phase Failed", reconcileNormalRemediationTestCase{
+			ExpectError:      false,
+			ExpectRequeue:    false,
+			RemediationPhase: infrav1.PhaseFailed,
+		}),
+	)
+
+	DescribeTable("Metal3Remediation marshal test",
+		func(tc marshallRemediationTestCase) {
+			nodeAnnotations, err := marshal(tc.Map)
+			if tc.Map == nil {
+				Expect(nodeAnnotations).To(Equal(""))
+			} else {
+				Expect(nodeAnnotations).ToNot(Equal(""))
+			}
+			Expect(err).To(BeNil())
 
-			DescribeTable("ReconcileNormal tests", func(tc reconcileNormalRemediationTestCase) {
-				m := setReconcileNormalRemediationExpectations(ctrl, tc)
-				res, err := remReconcile.reconcileNormal(context.TODO(), m)
+		},
+		Entry("Should return empty string",
```

Comment:
```suggestion
		Entry("Should return an empty string",
```

Reply:
Thank you @furkatgofurov7 for the review, fixes are done, kindly check again.